examples/pre_init.rs

```diff
@@ -0,0 +1,36 @@
+//! `cortex-m-rt` based program with a function run before RAM is initialized.
+
+#![deny(warnings)]
+#![no_main]
+#![no_std]
+
+#[macro_use(entry, exception, pre_init)]
+extern crate cortex_m_rt as rt;
+extern crate panic_semihosting;
+
+use rt::ExceptionFrame;
+
+pre_init!(disable_watchdog);
+
+unsafe fn disable_watchdog() {
+    // Do what you need to disable the watchdog.
+}
+
+// the program entry point
+entry!(main);
+
+fn main() -> ! {
+    loop {}
+}
+
+// the hard fault handler
```

Comment:
The `exception!` handlers here are not needed, as they are defaulted now.

Reply:
Ok, I can remove them. I simply copied and changed the code from the "minimal" example without looking too closely at everything else.